### DISCUSSION TITLE
Grain boundary length

### DISCUSSION THREAD

**Discussion by mugikumar** [mtex-toolbox/mtex#1782] [Ask Anything]
Hi developers,
               I have done the length calculation of high-angle and low-angle boundaries by mtex. I was getting the high angle boundaries value of 1.4944e15 micrometre (14.94cm), but in TSL, showing the same map value of 22.30 cm.
I have used the mtex code of
gb = [grains.boundary('indexed','indexed') grains('indexed').innerBoundary]
mori = gb.misorientation
making the condition for high angle boundaries
l = sum(gb(cond).segLength)

**Comment by lfgmorales** [comment]
Hi

I guess your definition of „grains“ is not the same on OIM and MTEX, but this is just an educated guess

Cheers


Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 29 Mar 2023, at 12:42, mugikumar ***@***.***> wrote:

﻿

Hi developers,
I have done the length calculation of high-angle and low-angle boundaries by mtex. I was getting the high angle boundaries value of 1.4944e15 micrometre (14.94cm), but in TSL, showing the same map value of 22.30 cm.
I have used the mtex code of
gb = [grains.boundary('indexed','indexed') grains('indexed').innerBoundary]
mori = gb.misorientation
making the condition for high angle boundaries
l = sum(gb(cond).segLength)

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1782>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN576NHZVPSK6Y7QGOTW6QG2RANCNFSM6AAAAAAWLZCMCE>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1782", "url": "https://github.com/mtex-toolbox/mtex/discussions/1782", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by mugikumar** [comment]
Hi Luiz Morales
             Thanks for your reply, which one is the correct one to use?

**Comment by lfgmorales** [comment]
Hi

There is not a „correct“ number, but for your measurements to be at least closer to each other, ypu have to define the object „grains“ in both approaches

Cheers


Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 29 Mar 2023, at 12:54, mugikumar ***@***.***> wrote:

﻿

Hi Luiz Morales
Thanks for your reply, which one is the correct one to use?

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1782#discussioncomment-5464396>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN5KO24JKEN4XIHTOH3W6QIFPANCNFSM6AAAAAAWLZCMCE>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1782#discussioncomment-5464396", "url": "https://github.com/mtex-toolbox/mtex/discussions/1782#discussioncomment-5464396", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by mugikumar** [comment]
Hi Luiz
          Thanks for your reply, in TSL-OIM grain definition are tolerance angle - 5 degree, grain minimum size - 2, CI>0.1 and MTEX grain definition are [grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',5*degree);
% EBSD data in small grains are removed
ebsd(grains(grains.grainSize < 2)) = [];
% Recalculate the grains from the remaining data ...
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',5*degree);
% ... and smooth the grain boundaries    
grains = smooth(grains,5);
I was getting different grain boundary length.
is both are same orelse which basis good for grain analysis?

**Reply by ralfHielscher** [reply]
How much does the length differ? Have you visually compared grains in both software? 

Ralf.

**Comment by mugikumar** [comment]
Hi ralf
 I got it both of the visually grains and boundaries are different when grain definition are same. i have attached the both of grain boundaries map and calc length. please correct me to get the good grain defn for calc length and map
cond1 = gb.misorientation.angle > 15*degree & ...
        gb.misorientation.angle < 65*degree;
cond = gb.misorientation.angle > 2*degree & ...
        gb.misorientation.angle < 15*degree;
l = sum(gb.segLength) this code is used for calc length in mtex
![mtex length ](https://user-images.githubusercontent.com/109027461/234480622-186e6a83-5f73-45cb-b523-72d3b662c2bd.jpg)
![grain boundary](https://user-images.githubusercontent.com/109027461/234480686-efd8032f-c8be-4130-85e6-430b99685497.jpg)
![length of gb](https://user-images.githubusercontent.com/109027461/234480693-4d9993c6-9f22-4662-a277-0cf077a4cc52.jpg)
![10 2_02](https://user-images.githubusercontent.com/109027461/234480740-31fe128e-2e45-40d9-9af3-769f19d9a0d0.png)

**Reply by ralfHielscher** [reply]
Some points:

1. your pictures are in a resolution that makes it very difficult to see anything. Maybe investigate only a smaller subregion first, to check whether there are differences. Maybe use the same color schema in both figures.

2. When computing grains with 
```matlab
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',5*degree)
```
you can not expect to have any grain boundary with misorientation angle smaller 5 degree. So you need to set this value to 2*degree.

3. In `grains.boundary` does not include inner grain boundary. Use `grains.innerBoundary` to access the inner boundaries.

4. smoothing grains reduces the stair casing effect and hence the length of the grain boundaries and should be applied.

Ralf.
